aptos-move/framework/aptos-stdlib/sources/simple_map.move

```diff
@@ -34,6 +34,18 @@ module aptos_std::simple_map {
         }
     }
 
+    public fun element_key<Key: store, Value: store>(elem: &Element<Key, Value>): &Key {
```

Comment:
the simple_map is sorted. so it is not iterable in its current implementation. The insert or delete will complete change the order

Reply:
> the simple_map is sorted. so it is not iterable in its current implementation. The insert or delete will complete change the order

The two added functions `borrow_at_index` and `borrow_at_index_mut` will prevent any `&mut` call on the map because the map is still being borrowed.

The `key` of the `Element` is not mutable by the function `element_key`, so the order of the element in `data` will not be changed.

The only way to allow modification of the map is to make a copy of the borrowed element, but I think that's fine?

Reply:
I don't see why there cannot be an add between two borrow_at_index call.

Reply:
> I don't see why there cannot be an add between two borrow_at_index call.

Because `borrow_at_index` borrows from the reference, `add` cannot take a mutable reference because of borrow checker.

```move
error[E07005]: invalid transfer of references
  ┌─ test-simplemap.move:6:9
  │
5 │         let v = borrow_at_index(&map, 1);
  │                 ------------------------ It is still being borrowed by this reference
6 │         add(&mut map, 10, 2);
  │         ^^^^^^^^^^^^^^^^^^^^ Invalid usage of reference as function argument. Cannot transfer a mutable reference that is being borrowed

{
  "Error": "Move compilation failed: Compilation error"
}
```

From below code:

```move
module test_map::test_map {
    use aptos_std::simple_map::{create, add, borrow_at_index, destroy_empty };
    public fun test_several() {
        let map = create<u64, u64>();
        let v = borrow_at_index(&map, 1);
        add(&mut map, 10, 2);
        let b = borrow_at_index(&map, 2);
        let _ = v;
        let _ = b;
        destroy_empty(map);
    }
}
```